backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java

```diff
@@ -71,4 +72,24 @@
   Page<TestEvent> findAll(Specification<TestEvent> searchSpec, Pageable p);
 
   long count(Specification<TestEvent> searchSpec);
+
+  @Query(
+      value =
+          "SELECT new gov.cdc.usds.simplereport.db.model.auxiliary.TestResultWithCount(te.result, COUNT(te)) "
+              + "FROM TestEvent te "
+              + "         LEFT JOIN TestEvent corrected_te ON corrected_te.priorCorrectedTestEventId = te.internalId "
+              + "WHERE te.facility.internalId = :facilityId AND COALESCE(te.dateTestedBackdate, te.createdAt) BETWEEN :startDate AND :endDate AND "
+              + "    te.correctionStatus = 'ORIGINAL' AND corrected_te.priorCorrectedTestEventId IS NULL "
+              + "GROUP BY te.result")
+  List<TestResultWithCount> countByResultInFacility(UUID facilityId, Date startDate, Date endDate);
+
+  @Query(
+      value =
+          "SELECT new gov.cdc.usds.simplereport.db.model.auxiliary.TestResultWithCount(te.result, COUNT(te)) "
+              + "FROM TestEvent te "
+              + "         LEFT JOIN TestEvent corrected_te ON corrected_te.priorCorrectedTestEventId = te.internalId "
+              + "WHERE te.organization.internalId = :orgId AND COALESCE(te.dateTestedBackdate, te.createdAt) BETWEEN :startDate AND :endDate AND "
+              + "    te.correctionStatus = 'ORIGINAL' AND corrected_te.priorCorrectedTestEventId IS NULL "
+              + "GROUP BY te.result")
+  List<TestResultWithCount> countByResultInOrganization(UUID orgId, Date startDate, Date endDate);
```

Comment:
Have you done a query plan on this. The test event table has a index on `facility_id` and `patient_id` but not `organization_id`. It might be more performant to do `countByResultInFacilities(List<UUID> facilityIds,...`

Reply:
I didn't spend too much time thinking about optimization of these, but will take a look.  Hitting an existing index would be nice.  It's also a good way to reduce these down to just 1 repo method.

Reply:
If we end up sticking with this query for the metrics, we should add these 2 indexes to improve performance of the query:

```plpgsql
-- for join
CREATE INDEX ix__test_event__prior_corrected_test_event_id ON test_event(prior_corrected_test_event_id);
-- for filter
CREATE INDEX "ix__test_event__fac_id-coal_test_date-test-corr_status" ON test_event(facility_id, COALESCE(date_tested_backdate, created_at), correction_status);
```

I think we should hold off on adding them until we've tested a little more and are sure this query is what we are going to go with for v1 of metrics.